src/main/java/com/example/bookstore/book/BookRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.example.bookstore.book;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+public interface BookRepository extends CrudRepository<Book, Integer> {
+
+    @Query("SELECT b FROM Book b WHERE LOWER(b.title) LIKE LOWER(concat('%', :query, '%')) OR LOWER(b.author) LIKE LOWER(concat('%', :query, '%')) OR LOWER(b.publisher) LIKE LOWER(concat('%', :query, '%'))")

```

Comment:
I am not familiar with why you need to specify the query like this.

Reply:
This is done like this so that the user can type in anything into the search bar, and it will check the title, author, and publisher to find a match. the LOWER is so that the query isn't case sensitive